mlrun/model_monitoring/db/tsdb/timescaledb/queries/timescaledb_results_queries.py

```diff
@@ -0,0 +1,583 @@
+# Copyright 2025 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from datetime import datetime, timedelta
+from typing import Optional, Union
+
+import pandas as pd
+import v3io_frames.client
+
+import mlrun
+import mlrun.common.schemas.model_monitoring as mm_schemas
+import mlrun.utils
+from mlrun.model_monitoring.db.tsdb.timescaledb.timescaledb_connection import (
+    Statement,
+)
+from mlrun.model_monitoring.db.tsdb.timescaledb.utils.timescaledb_dataframe_processor import (
+    TimescaleDBDataFrameProcessor,
+)
+from mlrun.model_monitoring.db.tsdb.timescaledb.utils.timescaledb_query_builder import (
+    TimescaleDBQueryBuilder,
+)
+
+
+class TimescaleDBResultsQueries:
+    """
+    Query class containing results and drift-related query methods for TimescaleDB.
+
+    Can be used as a mixin or standalone instance with proper initialization.
+    """
+
+    def __init__(
+        self,
+        project: Optional[str] = None,
+        connection=None,
```

Comment:
Connection should be a required parameter.
```suggestion
        connection: TimescaleDBConnection,
        project: Optional[str] = None,
```

Reply:
will fix